### PR TITLE
added archer OpenMP race detector and its deps

### PR DESCRIPTION
**Pull request by lee218llnl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/archer/package.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+##############################################################################
+
+from spack import *
+
+
+class Archer(Package):
+    """ARCHER, a data race detection tool for large OpenMP applications."""
+
+    homepage = "https://github.com/PRUNERS/ARCHER"
+    base_url = "https://github.com/PRUNERS/ARCHER/commit"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/archer/package.py]: What is `base_url`?

File: var/spack/repos/builtin/packages/llvm-openmp-ompt/package.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+##############################################################################
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/llvm-openmp-ompt/package.py]: You accidentally committed your vim swap file

File: var/spack/repos/builtin/packages/py-lit/package.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+       summarizing their results, and providing indication of failures. lit is
+       designed to be a lightweight testing tool with as simple a user
+       interface as possible."""
+
+    homepage = "https://pypi.python.org/pypi/lit"
+    url      = "https://pypi.python.org/packages/5b/a0/dbed2c8dfb220eb9a5a893257223cd0ff791c0fbc34ce2f1a957fa4b6c6f/lit-0.5.0.tar.gz"
+
+    version('0.5.0',  '8144660cc692be8fb903395a5f06564d')
+
+    extends('python')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-lit/package.py]: `extends('python')` is already in the `PythonPackage` base class, so you don't need this line.

**Comment by lee218llnl** [comment]
@adamjstewart, thanks for the reviews, I think I addressed all of your comments.

File: var/spack/repos/builtin/packages/archer/package.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    depends_on('ninja', type='build')
+    depends_on('llvm-openmp-ompt')
+
+    def install(self, spec, prefix):
+
+        with working_dir('spack-build', create=True):
+            cmake_args = std_cmake_args[:]
+            cmake_args.extend([
+                '-G',
+                'Ninja',
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/archer/package.py]: Can I be nitpicky and request `'-G'` and `'Ninja'` on the same line (but not in the same string)?

File: var/spack/repos/builtin/packages/archer/package.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                '-DCMAKE_C_COMPILER=clang',
+                '-DCMAKE_CXX_COMPILER=clang++',
+                '-DOMP_PREFIX:PATH=%s' %spec['llvm-openmp-ompt'].prefix,
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/archer/package.py]: `flake8` will complain about `%spec`

File: var/spack/repos/builtin/packages/llvm-openmp-ompt/package.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    depends_on('llvm+clang~gold')
+    depends_on('ninja', type='build')
+
+    def install(self, spec, prefix):
+
+        with working_dir('spack-build', create=True):
+            cmake_args = std_cmake_args[:]
+            cmake_args.extend([
+                '-G',
+                'Ninja',
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/llvm-openmp-ompt/package.py]: same request as above

**Comment by lee218llnl** [comment]
@tgamblin I pushed the changes you requested and it looks like it passed the checks.

**Comment by tgamblin** [comment]
@lee218llnl: Thanks!  LGTM.  Do we need to add any parallel support for `ninja` like we have for `make`?

**Comment by lee218llnl** [comment]
@tgamblin I'm not familiar with ninja, but I guess it will need parallel build support.

**Pull request merged**